examples/knative/templates/server.ps1

```diff
@@ -1,141 +1,156 @@
-if(${env:PORT}) {
-   $url = "http://*:${env:PORT}/"
-   $localUrl = "http://localhost:${env:PORT}/"
-} else {
-   $url = "http://*:8080/"
-   $localUrl = "http://localhost:8080/"
+if (${env:PORT}) {
+    $url = "http://*:${env:PORT}/"
+    $localUrl = "http://localhost:${env:PORT}/"
+}
+else {
+    $url = "http://*:8080/"
+    $localUrl = "http://localhost:8080/"
 }
 
 $serverStopMessage = 'break-signal-e2db683c-b8ff-4c4f-8158-c44f734e2bf1'
 
 . ./handler.ps1
 
 $backgroundServer = Start-ThreadJob {
-   param($url, $serverStopMessage)
-
-   Import-Module 'Microsoft.PowerShell.Utility'
-   Import-Module CloudEvents.Sdk
-
-   . ./handler.ps1
-
-   function Start-HttpCloudEventListener {
-   <#
-      .DESCRIPTION
-      Starts a HTTP Listener on specified Url
-   #>
-
-   [CmdletBinding()]
-   param(
-      [Parameter(
-         Mandatory = $true,
-         ValueFromPipeline = $false,
-         ValueFromPipelineByPropertyName = $false)]
-      [ValidateNotNull()]
-      [string]
-      $Url
-   )
-
-      $listener = New-Object -Type 'System.Net.HttpListener'
-      $listener.AuthenticationSchemes = [System.Net.AuthenticationSchemes]::Anonymous
-      $listener.Prefixes.Add($Url)
-
-      $cloudEvent = $null
-
-      try {
-         $listener.Start()
-         $context = $listener.GetContext()
-
-         try {
-            # Read Input Stream
-            $buffer = New-Object 'byte[]' -ArgumentList 1024
-            $ms = New-Object 'IO.MemoryStream'
-            $read = 0
-            while (($read = $context.Request.InputStream.Read($buffer, 0, 1024)) -gt 0) {
-               $ms.Write($buffer, 0, $read);
-            }
-            $bodyData = $ms.ToArray()
-            $ms.Dispose()
-
-            # Read Headers
-            $headers = @{}
-            for($i =0; $i -lt $context.Request.Headers.Count; $i++) {
-               $headers[$context.Request.Headers.GetKey($i)] = $context.Request.Headers.GetValues($i)
+    param($url, $serverStopMessage)
+
+    Import-Module 'Microsoft.PowerShell.Utility'
+    Import-Module CloudEvents.Sdk
+
+    . ./handler.ps1
```

Comment:
handler sourced twice?

Reply:
yes, the reason is this code runs in a separate PowerShell runspace initiated by `Start-ThreadJob`. PowerShell has the concept of runspaces running in different process threads. Runspaces do not share loaded PS code/modules so each one has to inialize what needs to ues.